presto-hive/src/main/java/io/prestosql/plugin/hive/util/Statistics.java

```diff
@@ -453,9 +454,13 @@ private static OptionalDouble getDoubleValue(ConnectorSession session, Type type
         return block.isNull(0) ? Optional.empty() : Optional.of(LocalDate.ofEpochDay(((SqlDate) type.getObjectValue(session, block, 0)).getDays()));
     }
 
-    private static OptionalLong getTimestampValue(Block block)
+    private static OptionalLong getTimestampValue(ConnectorSession session, Type type, Block block)
     {
-        return block.isNull(0) ? OptionalLong.empty() : OptionalLong.of(floorDiv(block.getLong(0, 0), MICROSECONDS_PER_SECOND));
+        if (block.isNull(0)) {
+            return OptionalLong.empty();
+        }
+        long epochMicros = ((SqlTimestamp) type.getObjectValue(session, block, 0)).getEpochMicros();
```

Comment:
Do not use `. getObjectValue`, it's supposed to be used for sending output to the client.
Please use `.getLong` or `.getObject`, depending on `type.isShort`

Reply:
I followed the pattern from method below.
```java
    private static Optional<BigDecimal> getDecimalValue(ConnectorSession session, Type type, Block block)
    {
        return block.isNull(0) ? Optional.empty() : Optional.of(((SqlDecimal) type.getObjectValue(session, block, 0)).toBigDecimal());
    }
```

What is the motivation to not use it. It makes the code simpler and the method has well defined semantics, right?

Reply:
Ok, so let me phrase this differently.
I think we generally avoid "the object representation" as I think it's supposed to be for the client procol.
Although i am aware javadoc doesn't say exactly this and we have some usages of the `getObjectValue` method outside of the communication impl.
So even, if it is the way I think it is, it is not documented.

Reply:
https://github.com/prestosql/presto/pull/5934